### PR TITLE
[DO NOT SQUASH] Merge razor-support feature branch

### LINKED ISSUE #7733: Update java plugin to report issue in .razor and .cshtml files

### LINKED ISSUE #7734: Enable code generated analysis for all analyzers
- [x] Enable code generated analysis by calling the [ConfigureGeneratedCodeAnalysis](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnostics.analysiscontext.configuregeneratedcodeanalysis)
- [x] Add Razor files exception to `GeneratedCodeRecognizer `

### LINKED ISSUE #7735: Update the Verifier to read Noncompliant comments from .cshtml and .razor files

### LINKED ISSUE #7737: Add ITs with .razor and .cshtml

### LINKED ISSUE #7739: Extend the test infrastructure to build and compile razor projects

### LINKED ISSUE #7818: Make it possible to disable razor analysis with the help of an environment variable
Make it possible to disable razor analysis with the help of an environment variable.

### LINKED ISSUE #7842: Test infrastructure: Programmatically set target framework
We should read and edit the `<TargetFramework>X</TargetFramework>` property within `EmptyProject.csproj`

### LINKED ISSUE #7843: Test infrastructure: Programmatically set language version
We should read and edit the `<LangVersion>X</LangVersion>` property within `EmptyProject.csproj`.

### LINKED ISSUE #7844: Test infrastructure: Programmatically add project references
We should read and edit the `<ProjectReference>X</ProjectReference>` property within `EmptyProject.csproj`.

### PR DESCRIPTION
**Pull request by csaba-sagi-sonarsource**
Fixes #7734
Fixes #7737 
Fixes #7739 
Fixes #7735 
Fixes #7733 
Fixes #7818 
Fixes #7842
Fixes #7843
Fixes #7844

### CODE REVIEW

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Tests/ProjectBuilderTest.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        f.Should().Throw<ArgumentException>();
+
+        f = () => EmptyVB.AddDocument("TestCases\\VariableUnused.unknown");
+        f.Should().Throw<ArgumentException>();
+    }
+
+    [TestMethod]
+    public void AddDocument_SupportsRazorFiles()
+    {
+        EmptyCS.AddDocument("TestCases\\UnusedPrivateMember.razor").FindDocument("UnusedPrivateMember.razor").Should().NotBeNull();
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Tests/ProjectBuilderTest.cs:43]: Empty line

File: its/src/test/java/com/sonar/it/csharp/Tests.java [hunk 2]
```diff
@@ -30,6 +30,8 @@
 import java.util.List;
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
+
```
Comment (cristian-ambrosini-sonarsource) [inline its/src/test/java/com/sonar/it/csharp/Tests.java:33]: Empty line (is it intentional?)

Comment (csaba-sagi-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/Tests.java:33]: Most probably not, I will fix it in an upcoming PR.

**Pull request merged**